apps/vmq_server/priv/lua_scripts/poll_main_queue.lua

```diff
@@ -0,0 +1,34 @@
+#!lua name=poll_main_queue
+
+--[[
+KEYS[1] = MainQueue
+ARGV[1] = maxMessagesCount
+]]
+
+local function poll_main_queue(KEYS, ARGV)
+    local mainQueue = KEYS[1]
+    local maxMsgCount = tonumber(ARGV[1])
+
+    local Res = {}
+    Res = redis.call('BLMPOP', 10, 1, mainQueue,  'RIGHT', 'COUNT', maxMsgCount)
```

Comment:
> Is BLMPOP supported on the current Memory store version?
It is not part of the blocked commands (ref: https://cloud.google.com/memorystore/docs/redis/product-constraints).

> Also, since BLMPOP is blocking, does this mean other read operations from the client calling BLMPOP will also be blocked?
No it does not block other commands on different connection.

Reply:
Just putting put this out for future reference. Using BLMPOP creates a few limitations:
- Same redis instance cannot be used in the future for subscription store and message store.
- Memory store is not supported.

Reply:
I dug deeper and found out that in lua blocking commands are same as non-blocking. So it does not matter actually.

>It is not part of the blocked commands (ref: https://cloud.google.com/memorystore/docs/redis/product-constraints).

By this I meant I did not find any resource that says memory store doesn't support blocking commands.

 >Same redis instance cannot be used in the future for subscription store and message store.

Why?

Reply:
It is blocked from the respect of client and not the main redis process. By blocking we mean that the client will wait for the process to return data from the queue.